demo_callbacks.py

```diff
@@ -0,0 +1,273 @@
+    return to_collapse_class + " collapsed" if to_collapse_class else "collapsed"
+
+
+@dash.callback(
+    Output("features", "max"),
+    Output("features", "value"),
+    Output("features", "marks"),
+    inputs=[
+        Input("dataset", "value"),
+    ],
+)
+def create_features_input(data_set: str) -> tuple[int, int, dict]:
+    """Updates the max, marks, and default value of the Number of Features slider on load and any time the data set is updated.
+
+    Args:
+        data_set: The data set selected.
+
+    Returns:
+        int: The maximum value for the number of features slider.
+        int: The default value for the number of features slider.
+        dict: The marks for the slider to display min/max values.
+    """
+    data = DataSet(data_set)
+    max_feat = data.n
+
+    value = round(max_feat / 2)
+
+    marks = {
+        1: "1",
+        max_feat: str(max_feat),
+    }
+
+    return max_feat, value, marks
+
+
+@dash.callback(
+    Output("input-graph", "figure"),
+    inputs=[
+        Input("input-graph", "hoverData"),
+        Input("dataset", "value"),
+        Input("input-redund", "value"),
+    ],
+)
+def draw_input_graph(hover_data: dict, data_set: str, show_red: bool) -> go.Figure:
+    """Runs on load and any time the data set is updated. Displays the features in the data, with a bar showing the relevance of each feature. If show_red is true, then hovering on each feature's column shows the correlation between that feature and every other feature.
+
+    Args:
+        hover_data: Input information about user mouse location.
+        data_set: The data set selected.
+        show_red: If we want to see redundancy.
+
+    Returns:
+        go.Figure: A Plotly figure object.
+    """
+
+    if ctx.triggered_id == "input-graph" and not show_red:
+        raise PreventUpdate
+
+    # Load the data set
+    data = DataSet(data_set)
+
+    return draw_bar_chart(hover_data, None, data, show_red)
+
+
+@dash.callback(
+    Output("output-graph", "figure"),
+    inputs=[
+        Input("output-graph", "hoverData"),
+        Input("results-redund", "value"),
+        Input("selected-features", "data"),
+        Input("soln-score", "data"),
+        State("dataset", "value"),
+    ],
+)
+def draw_output_graph(
+    hover_data: dict, show_red: bool, selected_features: list, soln_score: float, data_set: str
+) -> go.Figure:
+    """Runs when the optimization step is complete. Displays the same bar graph as on the "Input" tab, with selected features solid/heavily outlined and unselected features semi-transparent.
+
+    Args:
+        hover_data: Input information about user mouse location.
+        show_red: If we want to see redundancy.
+        selected_features: The features selected for the model.
+        soln_score: The accuracy score for the model using the selected features.
+        data_set: The data set selected.
+
+    Returns:
+        go.Figure: A Plotly figure object.
+    """
+
+    # Load the data set
```

Comment:
Same here as in the draw_input function, this will help with flickering

```suggestion
    """

    if ctx.triggered_id == "output-graph" and not show_red:
        raise PreventUpdate

    # Load the data set
```